bcda/cli.go

```diff
@@ -45,21 +46,79 @@ type cclfFileMetadata struct {
 	filePath  string
 }
 
-func importCCLF8(fileMetadata cclfFileMetadata) error {
-	if fileMetadata.filePath == "" {
-		return errors.New("file path (--file) must be provided")
+type cclfFileValidator struct {
+	totalRecordCount int
+	maxRecordLength  int
+}
+
+func importCCLF0(fileMetadata cclfFileMetadata) (map[string]cclfFileValidator, error) {
+	if (cclfFileMetadata{}) == fileMetadata {
+		err := errors.New("File CCLF0 not found")
+		log.Error(err)
+		return nil, err
+	}
+
+	file, err := os.Open(filepath.Clean(fileMetadata.filePath))
+	if err != nil {
+		log.Error(err)
+		return nil, err
+	}
+
+	log.Infof("File contains %s data for ACO %s at %s.\n", fileMetadata.env, fileMetadata.acoID, fileMetadata.timestamp)
+
+	const (
+		fileNumStart, fileNumEnd           = 0, 13
+		totalRecordStart, totalRecordEnd   = 35, 55
+		recordLengthStart, recordLengthEnd = 56, 69
+	)
+
+	var validator map[string]cclfFileValidator
+	sc := bufio.NewScanner(file)
+	for sc.Scan() {
+		b := sc.Bytes()
+		if len(bytes.TrimSpace(b)) > 0 {
+			filetype := string(bytes.TrimSpace(b[fileNumStart:fileNumEnd]))
+			if filetype == "CCLF8" || filetype == "CCLF9" {
+				if validator == nil {
+					validator = make(map[string]cclfFileValidator)
+				}
+				count, err := strconv.Atoi(string(bytes.TrimSpace(b[totalRecordStart:totalRecordEnd])))
+				if err != nil {
+					log.Error(err)
+					return nil, err
+				}
+				length, err := strconv.Atoi(string(bytes.TrimSpace(b[recordLengthStart:recordLengthEnd])))
+				if err != nil {
+					log.Error(err)
+					return nil, err
+				}
+				validator[filetype] = cclfFileValidator{totalRecordCount: count, maxRecordLength: length}
+			}
+		}
+	}
+
+	if len(validator) < 2 {
```

Comment:
Should we be more specific in checking that there are CCLF8 and CCLF9 entries in the map? Seems possible (though very unlikely) that CCLF0 could have two CCLF8 rows but no CCLF9 rows, for example, and we wouldn't realize we hadn't gotten CCLF9 info.

Reply:
yeah that's a good check, I'll add that 👍